src/emulator/remoteconfig/index.ts

```diff
@@ -0,0 +1,301 @@
+    this._templateWatcher = chokidar.watch(templatePath, { persistent: true, ignoreInitial: true });
+    this._templateWatcher.on("change", async () => {
+      // There have been some race conditions reported (on Windows) where reading the
+      // file too quickly after the watcher fires results in an empty file being read.
+      // Adding a small delay prevents that at very little cost.
+      await new Promise((res) => setTimeout(res, 5));
+
+      this._logger.logLabeled(
+        "BULLET",
+        "remoteconfig",
+        `Change detected, updating template for Remote Config...`
+      );
+      this.updateTemplateSource(templatePath);
+      this.loadTemplate(this._templateSource);
+    });
+
+    const server = this._app.listen(port, host);
+    this.destroyServer = utils.createDestroyer(server);
+  }
+
+  public loadTemplate(source?: Source): void {
+    if (!source) {
+      source = this._templateSource;
+    }
+    // TODO(kroikie): validate template before using it.
+    this._emulatorTemplate = RemoteConfigEmulator.prepareEmulatorTemplate(
+      JSON.parse(source!.files[0].content)
+    );
+  }
+
+  /**
+   * Update incoming template to be used by the emulator server.
+   * If necessary `!isEmulator` conditional values are added to
+   * each parameter.
+   *
+   * @param template A valid remote config template.
+   */
+  static prepareEmulatorTemplate(template: any): any {
+    const emulatorTemplate = cloneDeep(template);
+    const emulatorParameters = emulatorTemplate["parameters"] || {};
+    for (const parameterName of Object.keys(emulatorParameters)) {
+      const emulatorParameter = emulatorParameters[parameterName];
+      if (emulatorParameter.hasOwnProperty("conditionalValues")) {
+        // add is_emulator
+        const conditionalValues = emulatorParameter["conditionalValues"];
+        if (!conditionalValues.hasOwnProperty("!isEmulator")) {
+          conditionalValues["!isEmulator"] = Object.assign({}, emulatorParameter["defaultValue"]);
+        }
+      } else {
+        // add conditional value object
+        emulatorParameter["conditionalValues"] = {
+          // add is_emulator
+          "!isEmulator": Object.assign({}, emulatorParameter["defaultValue"]),
+        };
+      }
+      emulatorParameters[parameterName] = emulatorParameter;
+    }
+    return emulatorTemplate;
+  }
+
+  /**
+   * Checks if the given RemoteConfigTemplate object is a valid emulator update.
+   * The object must be a valid Remote Config template and have valid !isEmulator values.
+   *
+   * @param {any} template A RemoteConfigTemplate object to be validated.
+   *
+   * @returns {any} Object indicating the validity of template.
+   * The returned object contains two parameters `valid` and `msg`. `valid` is true if
+   * the template is valid and false otherwise. `msg` describes the reason for the `valid`
+   * value.
+   */
+  static validateRemoteConfigEmulatorTemplate(template: any): ValidTemplate {
```

Comment:
High level feedback: Right now, this seems to just return the first reason a RC template is invalid. I think it would be better to instead try validate the whole thing and return a list of all of the problems.

Instead of just returning every time you find an issue, consider having a array of validationMessages that you continually add to. Then, at the end, you can check the length of that array to see if the template is valid, and either join the error messages together or just return them as an array in ValidTemplate.